### PR TITLE
Mandy mod2 feedback

### PR DESCRIPTION
**Pull request by mandyWW**
Hi, I've updated the coach notes with feedback from how my module 1/2 teaching went and suggestions for improvements ready to discuss in the retro meeting on Tuesday. I've also made a few minor updates to the actual course notes. 

Overall, I thought the content/order of topics was excellent and very industry relevant. We managed 4 different languages (Kotlin, Java, JavaScript & Django) too so this demonstrated the topics were able to be taught in language independent manner.

### CODE REVIEW

File: notes/Module-2/2.2.5-Unit_2_Knowledge_quiz.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Coach Notes
+
+## Coach comments
+Please add comments below which will help future coaches delivering this topic or suggestions for improvement of the content.
+
+|**Coach name**|**Delivery date**|**Comments**|
+|--------------|-----------------|------------|
+|Mandy|13/05/21|The following questions need reviewing: 8 - I thought the answer was URL Encoding, not sure URI Encoding exists?, 14 - sneaker database is not accessible now with a key, 16 & 17 - my understanding of static testing is pre-compilation time i.e. when you use tools like Sonar or manual code reviews to run static analysis of code. What is described is to me, unit tests and integration tests.|
```
Comment (bmordan) [inline notes/Module-2/2.2.5-Unit_2_Knowledge_quiz.md:8]: Yes the sneaker database is now behind RapidAPI paywall :( would need to find another open API?

File: notes/Module-2/2.2.4-Integration_tests.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Coach Notes
+
+## Coach comments
+Please add comments below which will help future coaches delivering this topic or suggestions for improvement of the content.
+
+|**Coach name**|**Delivery date**|**Comments**|
+|--------------|-----------------|------------|
+|Mandy|13/05/21|My understanding of static tests is pre-compilation time i.e. when you use tools like Sonar or manual code reviews to run static analysis of code. Could we rephrase to talk about unit versus integration testing? |
```
Comment (bmordan) [inline notes/Module-2/2.2.4-Integration_tests.md:8]: This distinction between Static and Dymamic tests is drawn from the BCS software methodologies syllabus and will come up again in Mod 3. Static and Dymamic tests are different from Unit and Integration tests. Each of these terms will need explaining separately.

File: notes/Module-2/2.2.5-Unit_2_Knowledge_quiz.md (hunk 1 above)
Comment (bmordan) [reply notes/Module-2/2.2.5-Unit_2_Knowledge_quiz.md:8]: Screenshot 2021-05-25 at 11 09 52

Comment (bmordan) [reply notes/Module-2/2.2.5-Unit_2_Knowledge_quiz.md:8]: URI encoding is a thing

File: notes/Module-2/2.3.1-Hashing_Functions.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# Coach Notes
+
+## Coach comments
+Please add comments below which will help future coaches delivering this topic or suggestions for improvement of the content.
+
+|**Coach name**|**Delivery date**|**Comments**|
+|--------------|-----------------|------------|
+|Mandy|13/05/21|Hashing as a topic is great but the assignment was too difficult for my apprentices - maybe have as an extension task? Instead, I created [some slides](https://docs.google.com/presentation/d/1rol2aj8sEfGUrvCKCW4VovpZH56_D7-sbDJ7b9gzbIU/edit#slide=id.gd9c17e1e99_0_17) and gave them an assignment to create a password cracker of MD5/SHA non-salted passwords. This worked well as it instilled the importance of choosing strong passwords and demonstrated why salts and stronger algorithms like bcrypt should be used. We should ensure we discuss which hashing algorithms are considered secure & which are not (mentioned in my slides) in the notes|
```
Comment (bmordan) [inline notes/Module-2/2.3.1-Hashing_Functions.md:8]: Too difficult even with the starter code? https://multiverselearningproducts.github.io/swe/notes/mod1/wk1/day4.html Maybe an extension activity then. So your slideshow is folded into the 2.0. Mod 2?

File: notes/Module-2/2.3.4-Unit_3_Knowledge_Quiz.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Coach Notes
+
+## Coach comments
+Please add comments below which will help future coaches delivering this topic or suggestions for improvement of the content.
+
+|**Coach name**|**Delivery date**|**Comments**|
+|--------------|-----------------|------------|
+|Mandy|13/05/21|Kahoot questions need rewording and better to make it into an Applied Quiz to get stats|
```
Comment (bmordan) [inline notes/Module-2/2.3.4-Unit_3_Knowledge_Quiz.md:8]: I vote we drop Kahoot and do our quizzes on applied

File: notes/Module-2/2.4.3-Session_based_auth.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# Coach Notes
+
+## Coach comments
+Please add comments below which will help future coaches delivering this topic or suggestions for improvement of the content.
+
+|**Coach name**|**Delivery date**|**Comments**|
+|--------------|-----------------|------------|
+|Mandy|17/05/21|Basic Auth does not have the concept of a logout - you have to exit the browser to logout. Give REST APIS should be stateless, I am not sure this lesson is necessary. It can get quite complicated implementing sessions in Spring (https://www.javainuse.com/spring/springboot_session). I think we should just jump straight into OAuth as feedback from apprentices was they wanted more time on the Banking App and working with OAuth. See https://restfulapi.net/statelessness.  |
```
Comment (bmordan) [inline notes/Module-2/2.4.3-Session_based_auth.md:8]: Agree lets get them to the banking app sooner.

File: notes/Module-2/2.4.4-Introduction_to_OAuth.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# Coach Notes
+
+## Coach comments
+Please add comments below which will help future coaches delivering this topic or suggestions for improvement of the content.
+
+|**Coach name**|**Delivery date**|**Comments**|
+|--------------|-----------------|------------|
+|Mandy|18/05/21|Learning objectives need to be modified|
+|Mandy|18/05/21|Assignment is not sufficient - to make the concept of OAuth more real, I got my students to sign up to the Spotify API and use their "Try in Console" feature to see how OAuth tokens are used to protect resources. It's great as you can see example cURL requests, simulate invalid tokens, expired tokens etc and see the Authorization Bearer tokens get issued and their expiry times. I would like to see this part of an assignment (as currently the assignment seems to just be copying the existing sequence diagram?) and get the students to create the UML diagrams based of their experience of using the Spotify API and as described in the [Spotify Implicit Grant Flow](https://developer.spotify.com/documentation/general/guides/authorization-guide/#implicit-grant-flow)|
```
Comment (bmordan) [inline notes/Module-2/2.4.4-Introduction_to_OAuth.md:9]: This seems like a great improvement. Can you add this to the session?

File: notes/Module-2/2.4.6-OpenID_Connect.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+|Mandy Warren|19/05/2020|Text after "OpenID connect" a bit confusing and does "Summary" seems to refer to the assignment as it mentions banking?|
+|Mandy Warren|19/05/2020|There are 2 assignments here - better to have the Banking one as the end of unit assessment (instead of a quiz) so we have a way to assess their knowledge. I gave this as an individual assessment as a) the group were all working in different languages and b) they wanted to work at their own pace and c) I needed to have evidence for how individuals were progressing. Also, although the old notes mention to NEVER commit passwords to version control, the new notes don't. It worries me having gmail account details in the code/in GitHub/on a Cloud server which gets forgotten about hence I did not get my cohort to do the emailing. I also asked them to produce a sequence diagram (including the User, the Banking App, Auth0 and the BankingAPI) which illustrated the process of logging in and sending a token to access the Banking APIs. This really helped their understanding and we could validate the design before they started coding.|
```
Comment (bmordan) [inline notes/Module-2/2.4.6-OpenID_Connect.md:11]: You can make these individual assignment.

**Review by bmordan — COMMENTED** [review]
Thanks Mandy

File: notes/Module-2/2.2.5-Unit_2_Knowledge_quiz.md (hunk 1 above)
Comment (mandyWW) [reply notes/Module-2/2.2.5-Unit_2_Knowledge_quiz.md:8]: yes there is a JS function called encodeURI but in Java its URLEncoder and other sites all seem to refer to URLEncoding e.g. https://www.w3schools.com/tags/ref_urlencode.ASP. I think if we just remove  one of URL encoded or URI encoded we'll save any confusion :-)

File: notes/Module-2/2.4.3-Session_based_auth.md (hunk 5 above)
Comment (mandyWW) [reply notes/Module-2/2.4.3-Session_based_auth.md:8]: Dan wanted to keep this so have left for now but will keep in coach notes this is optional and can be skipped.

File: notes/Module-2/2.4.6-OpenID_Connect.md (hunk 7 above)
Comment (mandyWW) [reply notes/Module-2/2.4.6-OpenID_Connect.md:11]: done

File: notes/Module-2/2.4.4-Introduction_to_OAuth.md (hunk 6 above)
Comment (mandyWW) [reply notes/Module-2/2.4.4-Introduction_to_OAuth.md:9]: done

**Pull request merged**